Orange/widgets/visualize/utils/plotutils.py

```diff
@@ -0,0 +1,151 @@
+from AnyQt.QtWidgets import QScrollArea, QSizePolicy, QAction
+from AnyQt.QtCore import Qt
+from AnyQt.QtGui import QKeySequence
+
+from Orange.widgets import gui
+
+
+JITTER_SIZES = [0, 0.1, 0.5, 1, 2, 3, 4, 5, 7, 10]
+
+
+def create_scroll_area(master):
+    scroll = QScrollArea(parent=master.controlArea)
+    groupbox = gui.widgetBox(scroll, "")
+    scroll.setWidget(groupbox)
+    scroll.setWidgetResizable(True)
+    master.controlArea.layout().addWidget(scroll)
+    return groupbox
+
+
+class Jittering:
+    def __init__(self, box_plot, master, jitter_sizes=None):
+        self.jitter_sizes = jitter_sizes if isinstance(jitter_sizes, list) else JITTER_SIZES
+        self.box_plot = box_plot
+        self.master = master
+
+    def __call__(self, is_cb_jitter_cont=False):
+        gui.valueSlider(
+            widget=self.box_plot, master=self.master, value='graph.jitter_size',
+            label="Jittering: ", values=self.jitter_sizes, callback=self.master.reset_graph_data,
+            labelFormat=lambda x:
+            "None" if x == 0 else ("%.1f %%" if x < 1 else "%d %%") % x)
+        if is_cb_jitter_cont:
+            gui.checkBox(
+                widget=gui.indentedBox(widget=self.box_plot),
+                master=self.master, value="graph.jitter_continuous",
+                label="Jitter numeric values", callback=self.master.reset_graph_data)
+
+
+class Box:
+    def __init__(self, master, groupbox=None):
+        self.master = master
+        self.groupbox = groupbox
+
+
+class BoxAxisData(Box):
+    def __call__(self):
+        common_options = dict(
+            labelWidth=50, orientation=Qt.Horizontal, sendSelectedValue=True,
+            valueType=str)
+        box_axis_data = gui.vBox(self.groupbox, "Axis Data")
+        self.master.cb_attr_x = gui.comboBox(
+            box_axis_data, self.master, "attr_x", label="Axis x:", callback=self.master.update_attr,
+            model=self.master.xy_model, **common_options)
```

Comment:
These classes suppose a number of master's method names, e.g. `update_attr` and attributes, e.g. `xy_model`. This does not look right -- but perhaps I'd do the same. Not sure.